### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by MaxSchmide**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const result = [];
```
Comment (ilona-borkovska) [inline src/transformStateWithClones.js:11]: image

**Review by ilona-borkovska — APPROVED** [review]

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -7,7 +7,28 @@
...
+  actions.map(action => {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, { ...action.extraData });
+        break;
+      case 'removeProperties':
+        action.keysToRemove.map(key => delete stateCopy[key]);
+        break;
+      case 'clear':
+        Object.keys(stateCopy).forEach(key => delete stateCopy[key]);
```
Comment (anstsot) [inline src/transformStateWithClones.js:22]: you can just assign the empty object to clear the state
```suggestion
        stateCopy = {};
```

**Review by anstsot — APPROVED** [review]
Great job, approving:sparkles:

File: src/transformStateWithClones.js (hunk 2 above)
Comment (MaxSchmide) [reply src/transformStateWithClones.js:22]: copy is constant so it won't work as tests said

**Pull request closed without merging**